src/xrGame/ui/UIBoostStatesWnd.cpp

```diff
@@ -0,0 +1,176 @@
+#include "StdAfx.h"
+#include "UIBoostStatesWnd.h"
+#include "UIHelper.h"
+
+CUIBoostStatesWnd::CUIBoostStatesWnd() 
+    : CUIWindow(CUIBoostStatesWnd::GetDebugType()) 
+{
+    bHorizontal = true;
+    bInverse = false;
+    dx = 0.f;
+    dy = 0.f;
+    maxItem = 8;
+}
```

Comment:
I believe more modern style is better:
```suggestion
CUIBoostStatesWnd::CUIBoostStatesWnd() 
    : CUIWindow(CUIBoostStatesWnd::GetDebugType()),
    bHorizontal(true),
    bInverse(false),
    dx(0.f),
    dy(0.f),
    maxItem(8)
{
}
```
actually in modern c++ it's recommended to initialize in declarations:
```c++
class CUIBoostStatesWnd final : public CUIWindow
{
// ...
private:
    bool bHorizontal{ true };
    bool bInverse{ false };
    float dx{ 0.0f };
    float dy{ 0.0f };
    u8 maxItem{ 8 };
    xr_vector<EBoostParams> indBoostPos;
    xr_map<EBoostParams, CUIStatic*> indBoostState;
};
```